ql/cashflows/cpicoupon.cpp

```diff
@@ -103,8 +103,11 @@ namespace QuantLib {
 
 
     Date CPICashFlow::baseDate() const {
-        // you may not have a valid date
-        QL_FAIL("no base date specified");
+        if(IndexedCashFlow::baseDate() !=  Date()){
+            return IndexedCashFlow::baseDate();
+        } else {
```

Comment:
I would write it as
```suggestion
        Date base = IndexedCashFlow::baseDate();
        if (base != Date()) {
            return base;
        } else {
```
to avoid the double method call.  The compiler can probably figure it out, but still.

Reply:
agreed!